hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -407,4 +414,24 @@ public static OMDirectoryCreateRequest getInstance(
     }
     return new OMDirectoryCreateRequest(omRequest, bucketLayout);
   }
+
+  @RequestFeatureValidator(
+      conditions = ValidationCondition.CLUSTER_NEEDS_FINALIZATION,
+      processingPhase = RequestProcessingPhase.PRE_PROCESS,
+      requestType = Type.CreateDirectory
+  )
+  public static OMRequest disallowCreateDirectoryWithECReplicationConfig(
+      OMRequest req, ValidationContext ctx) throws ServiceException {
+    if (ctx.versionManager().getMetadataLayoutVersion()
+        < OMLayoutFeature.ERASURE_CODED_STORAGE_SUPPORT.layoutVersion()) {
+      if (req.getCreateDirectoryRequest().getKeyArgs()
+          .hasEcReplicationConfig()) {
```

Comment:
Actually it does not mean anything as we support only the bucket level default replication configuration.

There are two reasons why I have added it here:
1. we are on the server side, and anyone can send us a request, not just our client, so a validation on this is useful in theory... in practice it may not ever gonna happen
2. if anytime in the future we decide to support specifying EC replication on a directory level like in HDFS, then this would be already here.

If neither of these seems to be useful, I am open to just remove this check anytime.

On the other hand I changed this to isallowed and reverted the condition.

Reply:
I think we should leave it. The directory create is done with the same KeyArgs proto, which could have the ECReplicationConfig field or EC ReplicationType set. Even if it is ignored when processing the request and would have no effect, it's probably safer with no drawbacks to leave this validator in.